cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -326,42 +331,61 @@ private void createStreamFeeds(StreamConfig config) {
 
   @Override
   public void truncate(Id.Stream streamId) throws Exception {
-    doTruncate(getStreamBaseLocation(streamId));
+    doTruncate(streamId, getStreamLocation(streamId));
   }
 
   @Override
   public void drop(Id.Stream streamId) throws Exception {
-    // Same as truncate
-    truncate(streamId);
+    doDrop(streamId, getStreamLocation(streamId));
   }
 
-  private Location getStreamConfigLocation(Id.Stream streamId) throws IOException {
-    return getStreamBaseLocation(streamId).append(CONFIG_FILE_NAME);
+  /**
+   * Returns the location that points the config file for the given stream.
+   */
+  private Location getConfigLocation(Id.Stream streamId) throws IOException {
+    return getStreamLocation(streamId).append(CONFIG_FILE_NAME);
   }
 
-  // Constructs path: /.../<namespace>/streams/<streamName>, as expected by StreamUtils#getStreamIdFromLocation
-  private Location getStreamBaseLocation(Id.Stream streamId) throws IOException {
-    return getStreamsHomeLocation(streamId.getNamespace()).append(streamId.getName());
+  /**
+   * Returns the location for the given stream.
+   */
+  private Location getStreamLocation(Id.Stream streamId) throws IOException {
+    return getNamespaceLocation(streamId.getNamespace()).append(streamId.getName());
   }
 
-  private Location getStreamsHomeLocation(Id.Namespace namespace) throws IOException {
+  /**
+   * Returns the location for the given namespace that contains all streams belong to that namespace.
+   */
+  private Location getNamespaceLocation(Id.Namespace namespace) throws IOException {
```

Comment:
This might give a wrong idea, no? Its actually returning the streams directory under the namespace, and not the namespace location?